src/pages/studyView/charts/ChartContainer.tsx

```diff
@@ -83,9 +89,15 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
                 this.mouseInChart = false;
             }),
             handleDownloadDataClick: () => {
-                let firstLine = this.props.chartMeta.displayName + '\tCount'
-                fileDownload(firstLine + '\n' + this.plot.downloadData(), this.fileName);
-
+                if (this.props.downloadData) {
+                    this.props.downloadData().then(data => {
+                        if (data && data.length > 0) {
+                            fileDownload(data, this.fileName);
+                        }
+                    }).catch(() => {
+                        // TODO this.triggerDownloadError();
```

Comment:
Soon, in the supporting virtual study branch, there are two ways of throwing errors. Error popup or error banner on the top.